server/src/objects/player.ts

```diff
@@ -909,13 +912,23 @@ export class Player extends BaseGameObject {
                 this.curWeapIdx = this.weaponManager.lastWeaponIdx;
                 break;
             case GameConfig.Input.EquipOtherGun:
-                for (let i = 0; i < this.weapons.length; i++) {
-                    if (GameConfig.WeaponType[i] === "gun" &&
-                            this.weapons[i] !== this.weapons[this.curWeapIdx]) {
-                        this.curWeapIdx = i;
-                        break;
-                    }
+                // for (let i = 0; i < this.weapons.length; i++) {
+                //     if (GameConfig.WeaponType[i] === "gun" &&
+                //             this.weapons[i] !== this.weapons[this.curWeapIdx]) {
+                //         this.curWeapIdx = i;
+                //         break;
+                //     }
+                // }
+
+                // completely unreadable but optimized weapon swapping code since leia apparently values fewer lines of code over readability
+                if (this.curWeapIdx == 0 || this.curWeapIdx == 1) {
+                    const otherGunSlotIdx = this.curWeapIdx ^ 1;
+                    const isOtherGunSlotFull: number = +!!this.weapons[otherGunSlotIdx].type;//! ! converts string to boolean, + coerces boolean to number
```

Comment:
+!! 💀 

I want to join the refactor train, here is my attempt.

```js
case GameConfig.Input.EquipOtherGun:
  const weapIdx = this.curWeapIdx === 0 ? 1 : 0;
  this.curWeapIdx = weapIdx;
  break;
```

Reply:
This isn't correct, equipothergun doesn't only work if you have a gun selected and switch to the other gun slot. It also deals with melee which is why my original code is necessarily more complex than your re-factor.

Reply:
damn, I really liked that two-line solution, it should handle switching from throwables too? I gave it another try and edited my solution, please take a look.

Reply:
my solution is how original surviv's equipothergun input action worked, it never worked with throwables. I'm 99.9% sure my original code is the most efficient way to program this.

Reply:
maybe it's the most efficient, I can't judge that. but it is unreadable and difficult to parse, at least for me.

> it never worked with throwables.

interesting, but doing nothing when the throwable slot is selected doesn't make sense.

Reply:
> but it is unreadable and difficult to parse
yeah i did it like that intentionally since leia wants everything to be ultra optimized and values less lines over readability. My original solution worked the same with more if statements and was much clearer to understand and she deleted it.

> but doing nothing when the throwable slot is selected doesn't make sense
I agree, I asked people in the community whats supposed to happen in original surviv when you run the input with a throwable selected.